tests/test_conduit_cli.py

```diff
@@ -0,0 +1,179 @@
+
+    mocked_secret = boto3.client("secretsmanager").create_secret(
+        Name="/copilot/dbt-app/staging/secrets/POSTGRES", SecretString="secretivestring"
+    )
+    expected_arn = mocked_secret["ARN"]
+
+    assert get_postgres_secret_arn("dbt-app", "staging") == expected_arn
+
+
+def test_is_task_running_when_task_is_not_running(mocked_cluster):
+    """Given an ECS Cluster ARN string, is_task_running should return False when
+    the task is not running."""
+
+    assert not is_task_running(mocked_cluster["cluster"]["clusterArn"])
+
+
+@mock_ec2
+def test_is_task_running(mocked_cluster):
+    """Given an ECS Cluster ARN string, is_task_running should return True when
+    the task is running."""
+
+    # Create mocked ECS Cluster
+    mocked_ecs_client = boto3.client("ecs")
+    mocked_cluster_arn = mocked_cluster["cluster"]["clusterArn"]
+    # Create mocked EC2 instance
+    mocked_ec2_client = boto3.client("ec2")
+    mocked_ec2_images = mocked_ec2_client.describe_images(Owners=["amazon"])["Images"]
+    mocked_ec2_client.run_instances(ImageId=mocked_ec2_images[0]["ImageId"], MinCount=1, MaxCount=1)
+    mocked_ec2_instances = boto3.client("ec2").describe_instances()
+    mocked_ec2_instance_id = mocked_ec2_instances["Reservations"][0]["Instances"][0]["InstanceId"]
+    mocked_ec2 = boto3.resource("ec2")
+    mocked_ec2_instance = mocked_ec2.Instance(mocked_ec2_instance_id)
+    mocked_instance_id_document = json.dumps(ec2_utils.generate_instance_identity_document(mocked_ec2_instance))
+    # Attach mocked EC2 instance to the mocked ECS Cluster
+    mocked_ecs_client.register_container_instance(
+        cluster=mocked_cluster_arn, instanceIdentityDocument=mocked_instance_id_document
+    )
+    mocked_task_definition_arn = mocked_ecs_client.register_task_definition(
+        family="copilot-dbtunnel",
+        containerDefinitions=[
+            {"name": "test_container", "image": "test_image", "cpu": 100, "memory": 500, "essential": True}
+        ],
+    )["taskDefinition"]["taskDefinitionArn"]
+    mocked_ecs_client.run_task(cluster=mocked_cluster_arn, taskDefinition=mocked_task_definition_arn)
+
+    assert is_task_running(mocked_cluster_arn)
+
+
+@patch("os.system")
+@patch("commands.conduit_cli.is_task_running", return_value=True)
+def test_exec_into_task(is_task_running, system):
+    """Test that exec_into_task runs the `copilot task exec` command with
+    expected --app and --env flags."""
+
+    exec_into_task("dbt-app", "staging", "arn:random")
+
+    system.assert_called_once_with("copilot task exec --app dbt-app --env staging")
+
+
+@freeze_time("Jan 14th, 2020", auto_tick_seconds=60)
+def test_exec_into_task_timeout(capsys):
+    exec_into_task("dbt-app", "staging", "arn:random")
+
+    assert (
+        capsys.readouterr().out
+        == "Attempt to exec into running task timed out. Try again by running `copilot task exec --app dbt-app --env staging or check status of task in Amazon ECS console.\n"
```

Comment:
Has the exec into the task really timed out? Or is the task just not running?

Reply:
The way I've used freezegun means that it doesn't hit the logic where it checks whether the task is running. It just sees that `time.time() >= timeout` and so `connected` is never set to True. I could remove the freezegun logic and it would eventually timeout because the task isn't running, but I don't want the test to take 60 seconds

Reply:
So does the test start at `2020-01-14 00:00:00` and the timeout period starts then, after which it somehow zooms forward 60 seconds?

Reply:
yeah it's the `auto_tick_seconds` bit that zooms it forward

Reply:
🪄